addOns/automation/src/main/resources/org/zaproxy/addon/automation/resources/auto.yaml

```diff
@@ -0,0 +1,55 @@
+--- # OWASP ZAP automation configuration file, for more details see https://www.zaproxy.com/docs/(TBA)
+env:
+  contexts:
+    - name: bodgeit
+      url: http://localhost:8080/bodgeit/  # The top level url, and the only mandatory parameter in the file, everything under this url will be included
+      includePaths:                        # An optional list of regexes to include
+      excludePaths:                        # An optional list of regexes to exclude
+        - 'http://localhost:8080/bodgeit/logout'
+      authentication:                      # TBA - in time to cover all auth configs
+  parameters:
+    failOnError: true                  
+    failOnWarning: false               
+    progressToStdout: true            
+
+jobs:
+  - type: addOns                       # Any non standard add-ons to install
+    parameters:
+      updateAddOns: false              # Default: true
+    install:
+    uninstall:                         # Any standard add-ons to uninstall
+
+  - type: passiveScan-config           # The passive scanner will always run - this job defines the rules it will use
+    parameters:
+      maxAlertsPerRule: 10             # Int: Maximum number of alerts to raise per rule
+    rules:                             # Can be used to override default settings
+      - id: 2
+        desc: Private IP Disclosure    # Not used - just for documentation
+        threshold: high
+
+  - type: spider                       # The traditional spider - fast but doesnt handle modern apps so well
+    parameters:
+      warnIfFoundUrlsLessThan: 50
+      failIfFoundUrlsLessThan: 20
+      maxDuration: 2                   # The max time in minutes the spider will be allowed to run for, default: 0 unlimited
+      maxDepth: 1
+      parseComments: true              # Bool: 
+
+  - type: passiveScan-wait             # The passive scanner will always run - this job just waits for it to finish
+    parameters:
+      maxDuration: 5                   # The max time to wait for the passive scanner, default: 0 unlimited
```

Comment:
Seems strange that the file goes pscan > spider > pscan

Reply:
Yeah, but we need to configure the pscan rules, explore the app and then wait for the pscan to finish (or be timed out). Strange or not, thats the best way to automate ZAP :)